packages/mongo/mongo_driver.js

```diff
@@ -949,7 +949,23 @@ Cursor.prototype.observe = function (callbacks) {
 
 Cursor.prototype.observeChanges = function (callbacks) {
   var self = this;
+  var methods = [
+    'addedAt',
+    'added',
+    'changedAt',
+    'changed',
+    'removedAt',
+    'removed',
+    'movedTo'
+  ];
   var ordered = LocalCollection._observeChangesCallbacksAreOrdered(callbacks);
+
+  methods.forEach(function (method) {
+    if (callbacks[method] && typeof callbacks[method] == "function") {
+      callbacks[method] = Meteor.bindEnvironment(callbacks[method]);
```

Comment:
I started to look into this a bit more but the code gets complex quite quick and I don't feel I have time to dig into that right now so I'd say let's keep this PR down to just binding the environment for the callbacks in the best way possible and open an issue for discussing that a better error handling strategy would be.